timeseries/src/autogluon/timeseries/models/multi_window/multi_window_model.py

```diff
@@ -124,15 +128,19 @@ def _fit(
                 else:
                     num_refits_remaining = math.ceil(
                         (val_splitter.num_val_windows - window_index) / refit_every_n_windows
-                    )
+                    ) + 1
                     time_left_for_window = time_left / num_refits_remaining
 
             if refit_this_window:
                 model = self.get_child_model(window_index)
+                _train, _val = train_fold, val_fold
+                if self.strict_val:
+                    _val_splitter = ExpandingWindowSplitter(prediction_length=self.prediction_length, num_val_windows=1)
+                    _train, _val = list(_val_splitter.split(train_fold))[0]
                 model_fit_start_time = time.time()
                 model.fit(
-                    train_data=train_fold,
-                    val_data=val_fold,
+                    train_data=_train,
+                    val_data=_val,
                     time_limit=time_left_for_window,
                     **kwargs,
                 )
```

Comment:
These changes implement the `strict_val` logic: Instead of training with the given `train_fold`/`val_fold` splits, we split `train_fold` into `_train`/`_val` and train on those. This seems to work well, and in particular we don't need any changes outside of those and still get the correct OOF predictions, which is convenient.